tests/testthat/test-qenv_eval_code.R

```diff
@@ -0,0 +1,154 @@
+testthat::test_that("eval_code evaluates the code in the qenvs environment", {
+  q1 <- new_qenv(quote(iris1 <- iris), env = list2env(list(iris1 = iris)))
+  q2 <- eval_code(q1, quote(b <- nrow(iris1)))
+  testthat::expect_identical(get_var(q2, "b"), 150L)
+})
+
+testthat::test_that("eval_code doesn't have access to environment where it's called", {
+  a <- 1L
+  q1 <- new_qenv(quote(a <- 1), env = environment())
+  b <- 2L
+  testthat::expect_s3_class(
+    eval_code(q1, quote(d <- b)),
+    c("qenv.error", "try-error", "error", "condition")
+  )
+})
+
+testthat::test_that("@env in qenv is always a sibling of .GlobalEnv", {
+  q1 <- new_qenv()
+  testthat::expect_identical(parent.env(q1@env), parent.env(.GlobalEnv))
+
+  q2 <- eval_code(q1, quote(a <- 1L))
+  testthat::expect_identical(parent.env(q2@env), parent.env(.GlobalEnv))
+  q3 <- eval_code(q2, quote(b <- 2L))
+  testthat::expect_identical(parent.env(q3@env), parent.env(.GlobalEnv))
+})
+
+testthat::test_that("library have to be called separately before using function from package", {
+  q1 <- eval_code(new_qenv(), quote(library(checkmate)))
+  # library call adds package env in the search path just over .GlobalEnv
+  #   it means that q3@env before the call was a parent of .GlobalEnv but not after the call
+
+  q2 <- eval_code(q1, quote(assert_number(1)))
+  testthat::expect_identical(parent.env(q2@env), parent.env(.GlobalEnv))
+
+  detach("package:checkmate", unload = TRUE)
```

Comment:
Just confirming this test leaves the status of the checkmate package unchanged (we don't need a on.exit(detach("package:checkmate")) on line 29 to make sure that it is detached if something goes wrong in lines 32 and 33? @gogonzo